test/suites/integration/access_log_policy_test.go

```diff
@@ -0,0 +1,651 @@
+package integration
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+	"github.com/aws/aws-sdk-go/service/firehose"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	anv1alpha1 "github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/pkg/config"
+	"github.com/aws/aws-application-networking-k8s/pkg/model/core"
+	"github.com/aws/aws-application-networking-k8s/test/pkg/test"
+)
+
+var _ = Describe("Creating Access Log Policy", Ordered, func() {
+	const (
+		k8sResourceName          = "test-access-log-policy"
+		bucketName               = "k8s-test-lattice-bucket"
+		logGroupName             = "k8s-test-lattice-log-group"
+		deliveryStreamName       = "k8s-test-lattice-delivery-stream"
+		deliveryStreamRoleName   = "k8s-test-lattice-delivery-stream-role"
+		deliveryStreamRolePolicy = `{
+			"Version": "2012-10-17",
+			"Statement": [
+				{
+					"Effect": "Allow",
+					"Action": ["s3:PutObject", "s3:GetBucketLocation"],
+					"Resource": ["arn:aws:s3:::k8s-test-lattice-bucket/*"]
+				}
+			]
+		}`
+		deliveryStreamAssumeRolePolicy = `{
+			"Version": "2012-10-17",
+			"Statement": [
+				{
+					"Effect": "Allow",
+					"Principal": {
+						"Service": "firehose.amazonaws.com"
+					},
+					"Action": "sts:AssumeRole"
+				}
+			]
+		}`
+	)
+
+	var (
+		s3Client          *s3.S3
+		logsClient        *cloudwatchlogs.CloudWatchLogs
+		firehoseClient    *firehose.Firehose
+		iamClient         *iam.IAM
+		httpDeployment    *appsv1.Deployment
+		grpcDeployment    *appsv1.Deployment
+		httpK8sService    *corev1.Service
+		grpcK8sService    *corev1.Service
+		httpRoute         *gwv1beta1.HTTPRoute
+		grpcRoute         *gwv1alpha2.GRPCRoute
```

Comment:
> The new e2e test suite consists of 8 tests and adds about 10% to the total e2e test run time

If you just create one of httpRoute/grpcRoute and create 2 AccessLogSubscriptions on it can it save some time? Or any other way to save e2etest time? (e.g., just test CloudWatchLogGroup type accessLog for gateway)

Reply:
We need 1 of each source type and 1 of each destination type to confirm all the use-cases work, otherwise there is risk that our TargetRef identification or DestinationArn validation is broken without us knowing. I'll be adding Delete and Update e2e tests here in upcoming PRs, I think we will be able to squish some use-cases together into single tests to save time once that happens.